git.md

```diff
@@ -1,2 +1,139 @@
+two kinds of changes, that feature should likely include two commits.
+
+## 5. Explain how to test the effect of the changes you made
+
+Testing your code is important, particularly when sharing it with
+others. Automated tests are the best but not the only way to test the effects
+of the changes you made. However you did it, explain how you ensured that the
+code behaves as you expect, and how someone else may reproduce your results.
+
+## 6. Structure each commit message like an email
+
+Like an email, commit messages have a subject and an optional body. The subject
+should start with a line summarizing your changes (aim for 50 characters or
+less). Use the imperative, present tense -- for example, "Fix typo",
+not "Fixed typo" or "Fixes typo". The message body, if present,
+should follow a blank line; wrap around 70 characters; and explain the
+motivation for the change ("why", not "what" or "how").
+
+See also:
+
+* [Example of a commit message](https://github.com/2DegreesInvesting/resources/issues/74).
+* [Commit best practices](https://r-pkgs.org/git.html#commit-best-practices).
+
+## 7. Agree on a workflow
+
+Choose a workflow based on your team's needs and preferences. However you
+choose to work, just make sure to agree on a common workflow that everyone
+follows:
+
+GitHub workflow:
+
+This workflow is 2DII's default. If a project uses a different
+workflow, document prominently (e.g. in README) what that workflow is
+([example](https://github.com/github/gitignore#contributing-workflow)).
+
+* Uses a central repository. 
+* Uses a single long-lived branch: "main"; it never contains broken code.
+* Developers work on "feature" branches (never on the "main" branch), share
+them via pull requests, and peers review the changes before they become a
+part of the main codebase.
+* A new release is created by tagging a particular commit on the main branch.
+
+Gitflow workflow:
+
+* Like the GitHub workflow but uses two long-lived branches: "main"
+and "develop". The main branch stores the official release history,
+and the develop branch serves as an integration branch for features.
+
+A team may agree on custom conventions beyond the "vanilla" workflows listed
+above.  For example, you may agree to prefix a commit-title with "WIP",
+"FIXME", "Amend: " or to whatever to communicate with your team succinctly
+via a vocabulary you all share.
+
+See also: 
+
+* [Comparing workflows](https://www.atlassian.com/git/tutorials/comparing-workflows).
+
+## 8. Use tools to be more productive
+
+Choose the tools that help you work more comfortably, more productively,
+and with fewer errors.
+
+See also:
+
+* [Install a Git client](https://happygitwithr.com/git-client.html)
+* [oh-my-zsh: Unleash your terminal like never before](https://ohmyz.sh/).
+* [gh: Take GitHub to the command line](https://cli.github.com/).
+* [Git from RStudio](https://rstudio.com/resources/webinars/managing-part-2-github-and-rstudio/).
```

Comment:
These are all so much about personal preference that it seems antithetical to a guidelines document that is/should be making a few hard "rules" about how we work together. Sharing ideas about tools and elective practices is great, but guidelines are specifically about the things that are *not* elective, right?

Reply:
Right -- with the caveat that this is aspirational as people are likely to face situations that make strict applicatoin impossible and may need to make informed trade-offs.

Reply:
While this document avoids details on _how_ to do things, it does point to external resources. These entries, in particular, respond to a request by @MonikaFu and @catarinabrg -- who liked some kind of optional guidance which may save some folks considerable time and frustration. 

In 4134e8e I start the document with this:

> This document describes git practices we agree by consensus. It focuses on _what_ to do and _why_, and only recommends optional resources about _how_ to do it.

This clarifies that the linked resources are optional (i.e. an opinion rather than a guideline). @cjyetman, is this a reasonable trade-off between your comment and the request by @MonikaFu and @catarinabrg?

Also, this comment addresses previous discussion with @AlexAxthelm -- where I said this document focuses on _what_ and _why_ but that wasn't explicit anywhere in the document -- until now.